backend/experiment/rules/tests/test_congosamediff.py

```diff
@@ -0,0 +1,115 @@
+from django.test import TestCase
+
+from experiment.models import Experiment
+from participant.models import Participant
+from section.models import Playlist as PlaylistModel, Section, Song
+from session.models import Session
+from experiment.actions import Explainer, Trial, Final, Playlist as PlaylistAction
+
+from experiment.rules.congosamediff import CongoSameDiff
+
+
+class CongoSameDiffTest(TestCase):
+
+    @classmethod
+    def setUpTestData(self):
+        self.section_csv = (
+            "Dave,m1_contour,0.0,20.0,samediff/melody_1_contour.wav,practice_contour,m1\n"
+            "Dave,m1_interval,0.0,20.0,samediff/melody_1_interval.wav,practice_interval,m1\n"
+            "Dave,m1_same,0.0,20.0,samediff/melody_1_same.wav,same,m1\n"
+            "Dave,m1_scale,0.0,20.0,samediff/melody_1_scale.wav,scale,m1\n"
+        )
+        self.playlist = PlaylistModel.objects.create(name='CongoSameDiff')
+        self.playlist.csv = self.section_csv
+        self.playlist.update_sections()
+        self.participant = Participant.objects.create()
+        self.experiment = Experiment.objects.create(
+            name='CongoSameDiff',
+            slug='congosamediff',
+            rounds=4,
+        )
+        self.session = Session.objects.create(
+            experiment=self.experiment,
+            participant=self.participant,
+            playlist=self.playlist
+        )
+
+    def test_initializes_correctly(self):
+        congo_same_diff = CongoSameDiff()
+        assert congo_same_diff.ID == 'CONGOSAMEDIFF'
+        assert congo_same_diff.contact_email == 'aml.tunetwins@gmail.com'
+
+    def test_first_round(self):
+        congo_same_diff = CongoSameDiff()
+        experiment = Experiment(id=1, name='CongoSameDiff', slug='congosamediff', rounds=4)
+        actions = congo_same_diff.first_round(experiment)
+        assert len(actions) >= 1
+        assert isinstance(actions[0], PlaylistAction)
+        assert isinstance(actions[1], Explainer)
+
+    def test_next_round_final_round(self):
+        congo_same_diff, session = create_context('final')
+
+        session.get_next_round = lambda: 6
```

Comment:
Clever! Never occurred to me that this was a possibility to test what happens after a number of rounds have passed in unit tests.